src/main/webapp/app/exercises/shared/statistics/doughnut-chart.component.ts

```diff
@@ -28,44 +27,29 @@ export class DoughnutChartComponent implements OnChanges, OnInit {
 
     constructor(private router: Router) {}
 
-    // Chart.js data
-    doughnutChartType: ChartType = 'doughnut';
-    doughnutChartColors: any[] = ['limegreen', 'red'];
-    doughnutChartLabels: string[] = ['Done', 'Not Done'];
-    totalScoreOptions: object = {
-        cutoutPercentage: 75,
-        scaleShowVerticalLines: false,
-        responsive: false,
-        tooltips: {
-            backgroundColor: 'rgba(0, 0, 0, 1)',
-            callbacks: {
-                label(tooltipItem: any, data: any) {
-                    if (data && data['datasets'] && data['datasets'][0] && data['datasets'][0]['data']) {
-                        const value = data['datasets'][0]['data'][tooltipItem['index']];
-                        return '' + (value === -1 ? 0 : value);
-                    }
-                    return '';
-                },
-            },
-        },
-    };
-    doughnutChartData: CourseStatisticsDataSet[] = [
-        {
-            data: [0, 0],
-            backgroundColor: this.doughnutChartColors,
-        },
+    // ngx
+    ngxDoughnutData: any[] = [
+        { name: 'Done', value: 0 },
+        { name: 'Not done', value: 0 },
```

Comment:
I think for better usability/maintainability and typesafety, it's worth introducing an interface for the array. The data always consists of two properties label and value, correct? Why don't you use an interface such as:
```
interface ngxDoughnoutDataEntry: {
label: string;
value: number; // or double
}
```

Reply:
I added an interface in edf3ef8